src/DistributedLocking/ILockProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Cascade\DistributedLocking;
+
+interface ILockProvider
```

Comment:
Habit -- for whatever reason `IFoo` has always been my default (maybe Java conventions). I'm OK with either approach -- would like to hear other opinions, but will change to be consistent with existing code if no one speaks up.